### DISCUSSION TITLE
ArmV7 (RasPi) support

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#2680] [Ideas]
Is there any plan to support arm?
Lots of people have now powerful RasPi (8GB Ram) so I guess I makes sense to support it

**Comment by unixfox** [comment]
As long as python run on ARM, Searx will run on it. I've looked into it and python can run on ARM so Searx support ARM.
Maybe we don't have tutorials on how to install Searx on ARM but that's a different issue.

**Comment by virtadpt** [comment]
The installation process is the same.

**Comment by 3nprob** [comment]
While it runs fine on both architectures, searx won't make use of 8GB RAM on armv7 - as a 32-bit architecture it maxes out at 4GB per process. aarch64/arm64 is the way to go these days and IMO it makes little sense to support 32-bit arm these days, even if it runs fine. If you have a powerful SBC like the RPi 4B, you want to run a 64-bit OS on it. RaspiOS is 64-bit by default now I think.

Easiest way to get it to run is to build the docker image locally. So locally you can produce a 64-bit image with `make docker`.

You can even build a cross-architecture image from your x64 machine pretty easily with buildx/binfmt if you have that set up already. In `manage.sh` replace `docker build` with e.g. `docker buildx build --platform linux/arm64  --platform linux/amd64` to produce an image that can run on either amd64 and arm64.

EDIT: What I said aboive holds for searx itself, but the provided setup scripts for morty and filtron require some modifications to work out of the box on alternative architectures. There's an unmerged PR that seems to resolve that here: https://github.com/searx/searx/pull/2490/files

**Comment by allendema** [comment]
Searx is working on your mentioned config RPI 4B/8GB.
But as of now I have a private instance.

Which step is not quite working for you?
